### ISSUE TITLE
HTTP/2 documentation

### ISSUE THREAD

**Issue by janilcgarcia** [http4s/http4s#6840]
**TLDR: This is a request for documentation regarding the use of HTTP/2 with ember**

I couldn't find any place in the docs talking about HTTP/2 with ember-server. I actually found the example in the repo, but I believe it would be good to have documentation on HTTP/2, especially because I had to do a bit research to figure out HTTP/2 only works on most browser if served through TLS and I'm not even sure if /push-promise is necessary to be honest.

I'm putting this up here and I will try to document that myself after I get it working here, but I think I'm the most qualified person for the job considering I just barely figure out how to do it :stuck_out_tongue:

**Comment by armanbilge** [comment]
Although HTTP/2 does have some additional features (such as push promises), for the most part you can just "drop it in".

I.e. all you need to do is add `.withHttp2` in the Ember server builder. Then it will use HTTP/2 with any client that asks for it, and you will enjoy all the benefits. Nothing else needs to change :)

**Comment by janilcgarcia** [comment]
@armanbilge That is indeed the case, but most browser don't support HTTP/2 without TLS and I think a small section under Service explaining that and the usage of `.withHttp2` would be enough.

Good to know I don't need push-promises, I'm a bit lost with regards to HTTP/2, I'm trying to avoid WebSockets but I want to support a long-running request (I'm actually trying to integrate Sangria subscriptions to http4s + cats-effect IO).

I actually had another problem, but I'll post in Discussions because I'm not entirely sure if it's me or http4s.
